src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1562,11 +1567,22 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 			this._initDimensions();
 			this.xterm?.raw.resize(this._cols || Constants.DefaultCols, this._rows || Constants.DefaultRows);
 		}
-
 		const originalIcon = this.shellLaunchConfig.icon;
-		await this._processManager.createProcess(this._shellLaunchConfig, this._cols || Constants.DefaultCols, this._rows || Constants.DefaultRows, this._accessibilityService.isScreenReaderOptimized()).then(error => {
+		await this._processManager.createProcess(this._shellLaunchConfig, this._cols || Constants.DefaultCols, this._rows || Constants.DefaultRows, this._accessibilityService.isScreenReaderOptimized()).then(async error => {
 			if (error) {
-				this._onProcessExit(error, error.code === ShellIntegrationExitCode);
+				if (this._usedShellIntegrationInjection) {
+					await this.reuseTerminal({ ignoreShellIntegration: true });
```

Comment:
I think we need to add current shell launch config props too?

```suggestion
					await this.reuseTerminal({
						...this._shellLaunchConfig,
						ignoreShellIntegration: true
					});
```